test/parallel/test-cluster-prof-and-logfile.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+const common = require('../common');
+const tmpdir = require('../common/tmpdir');
+
+const fs = require('fs');
+const assert = require('assert');
+const cluster = require('cluster');
+
+const waitForEvent = (eventEmiter, event) =>
+  new Promise((resolve) => eventEmiter.once(event, resolve));
+
+function findLogFileWithName(name, dir) {
+  const data = fs.readdirSync(dir, 'utf8');
+  return data.filter((file) => file.includes(name));
+}
+
+async function shouldFillLogfileArg() {
+  tmpdir.refresh();
+
+  cluster.setupMaster({
+    exec: __dirname,
+    execArgv: ['--prof', '--logfile'],
+    cwd: tmpdir.path
+  });
+
+  cluster.fork();
+
+  cluster.once('exit', common.mustCall((worker) => {
```

Comment:
Do you need these `'exit'` handlers? Can't you perform these checks after you `await` the `'exit'` event?